src/test/java/com/karankumar/bookproject/backend/service/PublisherServiceTest.java

```diff
@@ -0,0 +1,135 @@
+
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.karankumar.bookproject.backend.service;
+
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.entity.Publisher;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.orm.jpa.JpaObjectRetrievalFailureException;
+
+import javax.transaction.Transactional;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@IntegrationTest
+@DisplayName("PublisherService should")
+public class PublisherServiceTest {
+
+    private final PublisherService publisherService;
+    private final BookService bookService;
+
+    @Autowired
+    PublisherServiceTest(PublisherService publisherService, BookService bookService){
+        this.publisherService = publisherService;
+        this.bookService = bookService;
+    }
+
+    @BeforeEach
+    public void setUp() {
+        resetServices();
+    }
+
+    void resetServices() {
+        bookService.deleteAll();
+        publisherService.deleteAll();
+    }
+
+
+    @Test
+    @DisplayName("Save a Publisher Properly")
+    void saveValidPublisher() {
+        // given
+        Publisher publisher = new Publisher("Test SavePublisher ");
+
+        // when
+        publisherService.save(publisher);
+
+        // then
+        assertThat(publisherService.findById(publisher.getId())).isNotNull();
+    }
+
+    @Test
+    @DisplayName("throw exception on an attempt to save a null Publisher")
+    void throwExceptionWhenSavingANullPublisher() {
+        assertThatExceptionOfType(NullPointerException.class)
+                .isThrownBy(() -> publisherService.save(null));
+    }
+
+    @Test
+    @DisplayName("throw exception on an attempt to delete a null Publisher")
+    void throwExceptionWhenDeletingANullPublisher() {
+        assertThatExceptionOfType(NullPointerException.class)
+                .isThrownBy(() -> publisherService.delete(null));
+    }
+
+    @Test
+    @Transactional
+    @DisplayName("Delete a Publisher")
+    void deleteExistingPublisher() {
+        // given
+        Publisher publisher = new Publisher("Test DeletePublisher ");
+        publisherService.save(publisher);
+        Long publisherId = publisher.getId();
+
+        // when
+        publisherService.delete(publisher);
+
+        // then
+        assertThatThrownBy(() -> publisherService.findById(publisherId)).isInstanceOf(
+                JpaObjectRetrievalFailureException.class);
+    }
```

Comment:
Is it not possible to delete a publisher?

Reply:
Yes, we can. 

What I'm trying here to delete the publisher and checking the existance of that publisher after that. It should throw an error .

Reply:
Ah, my bad. I should have looked closer at the assertion. Thanks for clarifying